source/ShipyardPanel.cpp

```diff
@@ -156,18 +156,33 @@ int ShipyardPanel::DetailWidth() const
 
 int ShipyardPanel::DrawDetails(const Point &center)
 {
-	string selectedItem = "No Ship Selected";
+	const string selectedItem = selectedShip ? selectedShip->ModelName() : "No Ship Selected";
 	const Font &font = FontSet::Get(14);
 	const Color &bright = *GameData::Colors().Get("bright");
 	const Color &dim = *GameData::Colors().Get("medium");
 	const Sprite *collapsedArrow = SpriteSet::Get("ui/collapsed");
 
-	int heightOffset = 20;
+	int heightOffset = 0;
+
+	if(planet && !planet->ShipyardDescription().empty())
+	{
+		Point descriptionPoint(center.X() - INFOBAR_WIDTH / 2 + 25, center.Y() + heightOffset);
+		description.SetAlignment(Alignment::JUSTIFIED);
+		description.SetWrapWidth(INFOBAR_WIDTH - 50);
+		description.SetFont(font);
+		description.Wrap(planet->ShipyardDescription());
+		heightOffset += description.Height();
+		description.Draw(descriptionPoint, bright);
+	}
+	// Draw this string representing the selected ship (if any), centered in the details side panel
+	Point selectedPoint(center.X() - INFOBAR_WIDTH / 2, center.Y() + heightOffset);
+	font.Draw({selectedItem, {INFOBAR_WIDTH - 20, Alignment::CENTER, Truncate::MIDDLE}},
+		selectedPoint, bright);
+	heightOffset += 20;
```

Comment:
this all seems to be the same code as for the outfitterpanel... dont they both inherit of smth? mb not and if not you dont need to change it it's fine I guess